changelog/front-commerce-3.5-2.30.mdx

```diff
@@ -0,0 +1,209 @@
+content on the subject:
+
+- [Understanding Server-Side Tracking](https://www.front-commerce.com/understanding-server-side-tracking/)
+- [Exploring Server-Side Tracking and Server-Sent Events](https://www.front-commerce.com/exploring-server-side-tracking-and-server-sent-events/)
+- [Client-side Tracking vs. Server-side Tracking: Navigating The Challenges and Solutions](https://www.front-commerce.com/client-side-tracking-vs-server-side-tracking-navigating-the-challenges-and-solutions/)
+- [Exploring Server-Side Tracking with Front-Commerce](https://www.front-commerce.com/exploring-server-side-tracking-with-front-commerce/)
+- [Understanding Server-Side Tracking for Enhanced Data Collection](https://www.front-commerce.com/understanding-server-side-tracking-for-enhanced-data-collection/)
+
+:::
+
+## Dynamic Routing in Front-Commerce 3.5
+
+Front-Commerce 3.5 introduces dynamic routing, a feature that brings more
+flexibility than ever to your Remix application. It allows you to customize your
+URL structure dynamically based on your data, without sacrificing performance or
+features.
+
+**This feature is unique in the Remix ecosystem and closes the gap with the 2.x
+versions of Front-Commerce.** We took the time to build it right, and we are
+looking forward to seeing how we can refine or improve the feature with your
+feedback.
+
+Here is a quick overview of what you can do with dynamic routing:
+
+- maintain the benefits from route based data fetching, code splitting and
+  mental model from Remix
+- support advanced URL structures matching. Example: `/category-:slug-:id`,
+  `/latest-:brand/seo-keyword-:filter.html`, …
+- generate URLs based on identifiers contributed across different backends such
+  as an E-Commerce platform, CMS service, …
+- have top-level URL segments that can render different routes based on the
+  path. Example: `/my-product` can render a product route while `/my-category`
+  or `/my-page` can render different routes
+- optimize the URL matching algorithm for performance by customizing URL
+  matchers execution plan based on your own context. Example: run fast matchers
+  in parallel, and slower ones in a second batch if needed
+- implement a redirection plan from your CMS. Example: redirect old URLs to new
+  ones when the title is updated
+- implement localized URLs for your routes. Example: `/contact-us` in English
+  and `/contactez-nous` in French
+- a simplified URL matchers API, that is only responsible for converting URLs to
+  route parameters
+
+We built dynamic routing to solve concrete use cases and challenges from
+advanced e-commerce projects. Please let us know any limitations you encounter
+and how we can improve the feature.
+
+Technically, dynamic routing is built on top of the Route API from Remix. It can
+be adopted progressively in your project. **We have updated our default routes
+and extensions to use dynamic route features when relevant … as you would
+expect!**
```

Comment:
I don't think the `Technically` brings much value.

```suggestion
Dynamic routing is built on top of the Route API from Remix. It can
be adopted progressively in your project. **We have updated our default routes
and extensions to use dynamic route features when relevant … as you would
expect!**
```

Reply:
I removed "technically"